From 15aa1518d0fa62cae085bcef67a50d09a07b1318 Mon Sep 17 00:00:00 2001
From: Paul Naoumenko <paulx.naoumenko@intel.com>
Date: Thu, 17 Oct 2013 18:30:16 +0200
Subject: [PORT FROM MAIN]wifi: Splitting defines in a separate header

BZ: 125764

Change-Id: If03742c448a1215329306038a36d19976363b0d8
Orig-Change-Id: I6e3a458875952e909d8752b3a12f1e74575e423c
Category: device enablement
Domain: CWS.WIFI-Common
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Quentin Casasnovas <quentinx.casasnovas@intel.com>
---
 wifi/wifi.c |   60 +-------------------------------------------
 wifi/wifi.h |   81 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 82 insertions(+), 59 deletions(-)
 create mode 100644 wifi/wifi.h

diff --git a/wifi/wifi.c b/wifi/wifi.c
index d52e45b..6b90820 100644
--- a/wifi/wifi.c
+++ b/wifi/wifi.c
@@ -14,30 +14,7 @@
  * limitations under the License.
  */
 
-#include <stdlib.h>
-#include <fcntl.h>
-#include <errno.h>
-#include <string.h>
-#include <dirent.h>
-#include <sys/socket.h>
-#include <unistd.h>
-#include <poll.h>
-#include <pthread.h>
-#include <assert.h>
-
-#include "hardware_legacy/wifi.h"
-#include "libwpa_client/wpa_ctrl.h"
-
-#define LOG_TAG "WifiHW"
-#include "cutils/log.h"
-#include "cutils/memory.h"
-#include "cutils/misc.h"
-#include "cutils/properties.h"
-#include "private/android_filesystem_config.h"
-#ifdef HAVE_LIBC_SYSTEM_PROPERTIES
-#define _REALLY_INCLUDE_SYS__SYSTEM_PROPERTIES_H_
-#include <sys/_system_properties.h>
-#endif
+#include "wifi.h"
 
 static struct wpa_ctrl *ctrl_conn;
 static struct wpa_ctrl *monitor_conn;
@@ -45,45 +22,10 @@ static struct wpa_ctrl *monitor_conn;
 /* socket pair used to exit from a blocking read */
 static int exit_sockets[2];
 
-extern int do_dhcp();
-extern int ifc_init();
-extern void ifc_close();
-extern char *dhcp_lasterror();
-extern void get_dhcp_info();
-extern int init_module(void *, unsigned long, const char *);
-extern int delete_module(const char *, unsigned int);
-void wifi_close_sockets();
-
 static char primary_iface[PROPERTY_VALUE_MAX];
 // TODO: use new ANDROID_SOCKET mechanism, once support for multiple
 // sockets is in
 
-#ifndef WIFI_DRIVER_MODULE_ARG
-#define WIFI_DRIVER_MODULE_ARG          ""
-#endif
-#ifndef WIFI_FIRMWARE_LOADER
-#define WIFI_FIRMWARE_LOADER		""
-#endif
-#define WIFI_TEST_INTERFACE		"sta"
-
-#ifndef WIFI_DRIVER_FW_PATH_STA
-#define WIFI_DRIVER_FW_PATH_STA		NULL
-#endif
-#ifndef WIFI_DRIVER_FW_PATH_AP
-#define WIFI_DRIVER_FW_PATH_AP		NULL
-#endif
-#ifndef WIFI_DRIVER_FW_PATH_P2P
-#define WIFI_DRIVER_FW_PATH_P2P		NULL
-#endif
-
-#ifndef WIFI_DRIVER_FW_PATH_PARAM
-#define WIFI_DRIVER_FW_PATH_PARAM	"/sys/module/wlan/parameters/fwpath"
-#endif
-
-#define WIFI_MODULE_43241_OPMODE	"/sys/module/bcm43241/parameters/op_mode"
-#define WIFI_MODULE_4334_OPMODE		"/sys/module/bcm4334/parameters/op_mode"
-#define WIFI_MODULE_4335_OPMODE		"/sys/module/bcm4335/parameters/op_mode"
-
 #define WIFI_DRIVER_LOADER_DELAY	1000000
 
 static const char IFACE_DIR[]           = "/data/system/wpa_supplicant";
diff --git a/wifi/wifi.h b/wifi/wifi.h
new file mode 100644
index 0000000..1a6f0bf
--- /dev/null
+++ b/wifi/wifi.h
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2008, The Android Open Source Project
+ * Copyright 2012-2013, Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef WIFI_H_
+# define WIFI_H_
+
+#include <stdlib.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <string.h>
+#include <dirent.h>
+#include <sys/socket.h>
+#include <unistd.h>
+#include <poll.h>
+#include <pthread.h>
+#include <assert.h>
+
+#include "hardware_legacy/wifi.h"
+#include "libwpa_client/wpa_ctrl.h"
+
+#define LOG_TAG "WifiHW"
+#include "cutils/log.h"
+#include "cutils/memory.h"
+#include "cutils/misc.h"
+#include "cutils/properties.h"
+#include "private/android_filesystem_config.h"
+#ifdef HAVE_LIBC_SYSTEM_PROPERTIES
+#define _REALLY_INCLUDE_SYS__SYSTEM_PROPERTIES_H_
+#include <sys/_system_properties.h>
+#endif
+
+#ifndef WIFI_DRIVER_MODULE_ARG
+#define WIFI_DRIVER_MODULE_ARG          ""
+#endif
+#ifndef WIFI_FIRMWARE_LOADER
+#define WIFI_FIRMWARE_LOADER		""
+#endif
+#define WIFI_TEST_INTERFACE		"sta"
+
+#ifndef WIFI_DRIVER_FW_PATH_STA
+#define WIFI_DRIVER_FW_PATH_STA		NULL
+#endif
+#ifndef WIFI_DRIVER_FW_PATH_AP
+#define WIFI_DRIVER_FW_PATH_AP		NULL
+#endif
+#ifndef WIFI_DRIVER_FW_PATH_P2P
+#define WIFI_DRIVER_FW_PATH_P2P		NULL
+#endif
+
+#ifndef WIFI_DRIVER_FW_PATH_PARAM
+#define WIFI_DRIVER_FW_PATH_PARAM	"/sys/module/wlan/parameters/fwpath"
+#endif
+
+#define WIFI_MODULE_43241_OPMODE	"/sys/module/bcm43241/parameters/op_mode"
+#define WIFI_MODULE_4334_OPMODE		"/sys/module/bcm4334/parameters/op_mode"
+#define WIFI_MODULE_4335_OPMODE		"/sys/module/bcm4335/parameters/op_mode"
+
+extern int do_dhcp();
+extern int ifc_init();
+extern void ifc_close();
+extern char *dhcp_lasterror();
+extern void get_dhcp_info();
+extern int init_module(void *, unsigned long, const char *);
+extern int delete_module(const char *, unsigned int);
+void wifi_close_sockets();
+
+#endif /* !WIFI_H_ */
-- 
1.7.4.1

